app/src/main/java/com/example/admin/lab_50_82/MultithreadingActivity.java

```diff
@@ -0,0 +1,216 @@
+package com.example.admin.lab_50_82;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v7.app.ActionBarActivity;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+
+public class MultithreadingActivity extends ActionBarActivity {
+
+    private static final String TAG = "Multithreading";
+    private Button launchAsyncTask_;
+    private Button launchThread_;
+    private Button launchThreadViaHandler_;
+    private ListView listView_;
+    private ArrayAdapter<String> adapter_;
+    private ArrayList<String> list_;
+    private Handler handler_;
+    private int j_;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_multithreading);
+
+        launchAsyncTask_ = (Button) findViewById(R.id.add_in_async_task);
+        launchThread_ = (Button) findViewById(R.id.add_in_thread_directly);
+        launchThreadViaHandler_ = (Button) findViewById(R.id.add_in_thread_via_handler);
+
+        String[] values = new String[]{getResources().getString(R.string.item_label)};
+        listView_ = (ListView) this.findViewById(R.id.list_view);
+
+        list_ = new ArrayList<String>(Arrays.asList(values));
+        adapter_ = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, list_);
+
+        listView_.setAdapter(adapter_);
+
+        handler_ = new Handler() {
+            public void handleMessage(android.os.Message message) {
+                updateAdapter(message.what);
+                if (isItTenthItem(message.what)) {
+                    onOffButtons(true);
+                }
+            };
+        };
+
+        launchAsyncTask_.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                list_.clear();
+                list_.add(getResources().getString(R.string.added_label));
+                adapter_.notifyDataSetChanged();
+                new addItemViaAsyncTask().execute();
+                onOffButtons(false);
```

Comment:
Suppose that you should disable buttons at first, not at the end, then start doing list clearing and so on.